registry/storage/signaturestore.go

```diff
@@ -47,33 +52,41 @@ func (s *signatureStore) Get(dgst digest.Digest) ([][]byte, error) {
 			defer wg.Done()
 			context.GetLogger(s.ctx).
 				Debugf("fetching signature from %q", sigPath)
-			p, err := s.blobStore.linked(sigPath)
-			if err != nil {
+
+			r := result{index: idx}
+			if p, err := s.blobStore.linked(sigPath); err != nil {
 				context.GetLogger(s.ctx).
 					Errorf("error fetching signature from %q: %v", sigPath, err)
-
-				// try to send an error, if it hasn't already been sent.
-				select {
-				case errCh <- err:
-				default:
-				}
-
-				return
+				r.err = err
+			} else {
+				r.signature = p
 			}
-			signatures[idx] = p
+
+			ch <- r
 		}(i, sigPath)
 	}
-	wg.Wait()
-
-	select {
-	case err := <-errCh:
-		// just return the first error, similar to single threaded code.
-		return nil, err
-	default:
-		// pass
+	done := make(chan struct{})
+	go func() {
+		wg.Wait()
+		close(done)
+	}()
+
+	// aggregrate the results
+	signatures := make([][]byte, len(signaturePaths))
+loop:
+	for {
+		select {
+		case result := <-ch:
+			signatures[result.index] = result.signature
+			if result.err != nil {
+				err = result.err // only set the first one.
```

Comment:
Is this comment intended as a TODO or was the previous line intended `if result.err != nil && err == nil `?